File: app/src/main/java/modernjava/ex01.java
package modernjava;

public class ex01 {
    static class Box<T> {
        private T value;

        void set(T value) {
            this.value = value;
        }

        T get() {
            return value;
        }
    }

    public static void main(String[] args) {
        Box<String> box = new Box<>();
        box.set("generic value");
        System.out.println(box.get());
    }
}

File: app-kotlin/src/main/kotlin/modernjava/ex01.kt
package modernjava

class ex01 {
    class Box<T> {
        private var value: T? = null

        fun set(value: T) {
            this.value = value
        }

        fun get(): T? = value
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val box = Box<String>()
            box.set("generic value")
            println(box.get())
        }
    }
}
